IdentityCore/src/cache/accessor/MSIDLegacyTokenCacheAccessor.m

```diff
@@ -179,24 +179,43 @@ - (MSIDRefreshToken *)getRefreshTokenWithAccount:(MSIDAccountIdentifier *)accoun
                                          context:(id<MSIDRequestContext>)context
                                            error:(NSError **)error
 {
-    MSID_LOG_VERBOSE(context, @"(Legacy accessor) Get refresh token with authority %@, clientId %@, familyID %@", configuration.authority, configuration.clientId, familyId);
-    MSID_LOG_VERBOSE_PII(context, @"(Legacy accessor) Get refresh token with authority %@, clientId %@, familyID %@, account %@", configuration.authority, configuration.clientId, familyId, account.homeAccountId);
+    return [self getRefreshableTokenWithAccount:account
+                                       familyId:familyId
+                                 credentialType:MSIDRefreshTokenType
+                                  configuration:configuration
+                                        context:context
+                                          error:error];
+}
+
+- (MSIDRefreshToken *)getRefreshableTokenWithAccount:(MSIDAccountIdentifier *)account
+                                            familyId:(NSString *)familyId
+                                      credentialType:(MSIDCredentialType)credentialType
+                                       configuration:(MSIDConfiguration *)configuration
+                                             context:(id<MSIDRequestContext>)context
+                                               error:(NSError **)error
+{
+    MSID_LOG_VERBOSE(context, @"(Legacy accessor) Get token %@ with authority %@, clientId %@, familyID %@", [MSIDCredentialTypeHelpers credentialTypeAsString:credentialType], configuration.authority, configuration.clientId, familyId);
+    MSID_LOG_VERBOSE_PII(context, @"(Legacy accessor) Get token %@ with authority %@, clientId %@, familyID %@, account %@", [MSIDCredentialTypeHelpers credentialTypeAsString:credentialType], configuration.authority, configuration.clientId, familyId, account.homeAccountId);
+    
+    if (credentialType!=MSIDRefreshTokenType && credentialType!=MSIDPrimaryRefreshTokenType) return nil;
 
-    MSIDRefreshToken *refreshToken = [self getLegacyRefreshTokenForAccountImpl:account
-                                                                      familyId:familyId
-                                                                 configuration:configuration
-                                                                       context:context
-                                                                         error:error];
+    MSIDRefreshToken *refreshToken = [self getLegacyRefreshableTokenForAccountImpl:account
```

Comment:
There's also difference in how ADAL stored PRTs. If I remember correctly, it had a hardcoded GUID that it was using to identify PRTs that it used instead of client ID. Since we now have an explicit type to differentiate primary refresh tokens, we don't need to do that. If we're going to use broker clientId for new PRT cache, legacy accessor should still use the GUID that previous broker used.

Reply:
Make sense, let me change it :)

Reply:
addressed :)